runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java

```diff
@@ -684,6 +682,135 @@ private UnionCoder createUnionCoder(Collection<TaggedPValue> taggedCollections)
+      extends FlinkStreamingPipelineTranslator.StreamTransformTranslator<
+      SplittableParDo.ProcessElements<InputT, OutputT, RestrictionT, TrackerT>> {
+
+    @Override
+    public void translateNode(
+        SplittableParDo.ProcessElements<InputT, OutputT, RestrictionT, TrackerT> transform,
+        FlinkStreamingTranslationContext context) {
+
+      SplittableParDo.ProcessFn<InputT, OutputT, RestrictionT, TrackerT> doFn =
+          transform.newProcessFn(transform.getFn());
+
+      // we assume that the transformation does not change the windowing strategy.
+      WindowingStrategy<?, ?> windowingStrategy =
+          context.getInput(transform).getWindowingStrategy();
+
+      List<TaggedPValue> outputs = context.getOutputs(transform);
+
+      Map<TupleTag<?>, Integer> tagsToLabels =
+          transformTupleTagsToLabels(transform.getMainOutputTag(), outputs);
+
+      List<PCollectionView<?>> sideInputs = transform.getSideInputs();
+
+      SingleOutputStreamOperator<RawUnionValue> unionOutputStream;
+
+      @SuppressWarnings("unchecked")
+      PCollection<
+          KeyedWorkItem<String, ElementAndRestriction<InputT, RestrictionT>>> inputPCollection =
+          (PCollection<KeyedWorkItem<String, ElementAndRestriction<InputT, RestrictionT>>>)
+              context.getInput(transform);
+
+      Coder<
+          WindowedValue<KeyedWorkItem<String,
+              ElementAndRestriction<InputT, RestrictionT>>>> inputCoder =
+          context.getCoder(inputPCollection);
+
+      DataStream<WindowedValue<KeyedWorkItem<String,
+          ElementAndRestriction<InputT, RestrictionT>>>> inputDataStream =
+          context.getInputDataStream(context.getInput(transform));
+      Coder keyCoder = StringUtf8Coder.of();
+
+      if (sideInputs.isEmpty()) {
+        SplittableDoFnOperator<
+            InputT, OutputT, RawUnionValue, RestrictionT, TrackerT> doFnOperator =
+            new SplittableDoFnOperator<>(
+                doFn,
+                inputCoder,
+                transform.getMainOutputTag(),
+                transform.getSideOutputTags().getAll(),
+                new DoFnOperator.MultiOutputOutputManagerFactory(tagsToLabels),
+                windowingStrategy,
+                new HashMap<Integer, PCollectionView<?>>(), /* side-input mapping */
+                Collections.<PCollectionView<?>>emptyList(), /* side inputs */
+                context.getPipelineOptions(),
+                keyCoder);
+
+        UnionCoder outputUnionCoder = createUnionCoder(outputs);
+
+        CoderTypeInformation<RawUnionValue> outputUnionTypeInformation =
+            new CoderTypeInformation<>(outputUnionCoder);
+
+        unionOutputStream = inputDataStream
+            .transform(transform.getName(), outputUnionTypeInformation, doFnOperator);
+
+      } else {
+        unionOutputStream = null;
+      }
+
+      SplitStream<RawUnionValue> splitStream = unionOutputStream
```

Comment:
It definitely is, since I basically copied the translation for `ParDo.BoundMulti` and replaced instantiations of `DoFnOperator` by `SplittableDoFnOperator`. 😉 

That's one of the places I though about when mentioning "code cleanups".